cryptography/src/commonMain/kotlin/com/wire/kalium/cryptography/exceptions/ProteusException.kt

```diff
@@ -18,9 +18,20 @@
 
 package com.wire.kalium.cryptography.exceptions
 
-class ProteusException(message: String?, val code: Code, cause: Throwable? = null) : Exception(message, cause) {
+class ProteusException(message: String?, val code: Code, val intCode: Int, cause: Throwable? = null) : Exception(message, cause) {
 
-    constructor(message: String?, code: Int, cause: Throwable? = null) : this(message, fromNativeCode(code), cause)
+    constructor(message: String?, code: Code, cause: Throwable? = null) : this(
+        message,
+        code,
+        toProteusCode(code),
```

Comment:
can we avoid mapping from int code to proteus code and then back to code, by using the int code constructor and then map from int to Code

Reply:
Actually we can't. 
If you check mappers, you'll find out that `IntCode` that comes from native might be `1,2,3...15`, on that other hand `IntCode` from Proteus: `201, 501, 310, 3, 303...` so these are different ints :)

Reply:
it is fine we can just take the native code as it is and store it, this will always be correct, while over time the mapping can be wrong